modules/webgl/src/classes/vertex-array.js

```diff
@@ -215,7 +215,12 @@ export default class VertexArray {
   unbindBuffers() {
     this.vertexArrayObject.bind(() => {
       if (this.elements) {
-        this.setElementBuffer(null);
+        this.clearDrawParams();
+
+        // Update vertexArray immediately if we have our own array
+        if (!this.vertexArrayObject.isDefaultArray) {
```

Comment:
Isn't this check already done inside `setElementBuffer`?

Reply:
The problem is that check is after `this.elements` get reassigned here: https://github.com/uber/luma.gl/blob/f474db152f12126d40e3a567a8b752dc8b050b64/modules/webgl/src/classes/vertex-array.js#L144

Reply:
OK. Just so we are aligned on the design. This is my picture:
- There is always multiple `VertexArray` instances. 
- They can optionally have their own underlying `VertexArrayObject` instances, or share a single `VertexArrayObject`.
- `VertexArray.setElementBuffer` is an operation on the unique `VertexArray` not the `VertexArrayObject.
- So normally it is correct to always update the `VertexArray`, and only update the `vertexArrayObject` if it is currently selected (though we do it at bind and unbind time only).

Conclusion: If we don't want to update the `VertexArray` itself, we should not call its methods, but instead some method on `VertexArrayObject`, e.g. `VertexArrayObject.setElementBuffer`?